test/integration/build_to_secrets_test.go

```diff
@@ -0,0 +1,512 @@
+			Expect(buildObject.Status.Reason).To(Equal("Succeeded"))
+
+			// delete a secret
+			Expect(tb.DeleteSecret(buildObject.Spec.Output.SecretRef.Name)).To(BeNil())
+
+			// assert that the validation happened one more time
+			buildObject, err = tb.GetBuild(buildName)
+			Expect(err).To(BeNil())
+			Expect(buildObject.Status.Registered).To(Equal(corev1.ConditionTrue))
+		})
+
+		It("should not validate when a missing secret is recreated without annotation", func() {
+			// populate Build related vars
+			buildName := BUILD + tb.Namespace
+			buildObject, err = tb.Catalog.LoadBuildWithNameAndStrategy(
+				buildName,
+				STRATEGY+tb.Namespace,
+				[]byte(test.BuildCBSMinimalWithFakeSecret),
+			)
+			Expect(err).To(BeNil())
+
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+			// wait until the Build finish the validation
+			buildObject, err := tb.GetBuildTillValidation(buildName)
+			Expect(err).To(BeNil())
+			Expect(buildObject.Status.Registered).To(Equal(corev1.ConditionFalse))
+			Expect(buildObject.Status.Reason).To(Equal(fmt.Sprintf("secret %s does not exist", buildObject.Spec.Output.SecretRef.Name)))
+
+			sampleSecret := tb.Catalog.SecretWithoutAnnotation(buildObject.Spec.Output.SecretRef.Name, buildObject.Namespace)
+
+			// generate resources
+			Expect(tb.CreateSecret(sampleSecret)).To(BeNil())
+
+			// // assert that the validation happened one more time
+			buildObject, err = tb.GetBuildTillRegistration(buildName, corev1.ConditionFalse)
+			Expect(err).To(BeNil())
+			Expect(buildObject.Status.Registered).To(Equal(corev1.ConditionFalse))
+			Expect(buildObject.Status.Reason).To(Equal(fmt.Sprintf("secret %s does not exist", buildObject.Spec.Output.SecretRef.Name)))
+
+		})
+
+		It("should validate when a missing secret is recreated with annotation", func() {
+			// populate Build related vars
+			buildName := BUILD + tb.Namespace
+			buildObject, err = tb.Catalog.LoadBuildWithNameAndStrategy(
+				buildName,
+				STRATEGY+tb.Namespace,
+				[]byte(test.BuildCBSMinimalWithFakeSecret),
+			)
+			Expect(err).To(BeNil())
+
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+			// wait until the Build finish the validation
+			buildObject, err := tb.GetBuildTillValidation(buildName)
+			Expect(err).To(BeNil())
+			Expect(buildObject.Status.Registered).To(Equal(corev1.ConditionFalse))
+			Expect(buildObject.Status.Reason).To(Equal(fmt.Sprintf("secret %s does not exist", "fake-secret")))
+
+			sampleSecret := tb.Catalog.SecretWithoutAnnotation(buildObject.Spec.Output.SecretRef.Name, buildObject.Namespace)
+
+			// generate resources
+			Expect(tb.CreateSecret(sampleSecret)).To(BeNil())
+
+			// we modify the annotation so automatic delete does not take place
+			data := []byte(fmt.Sprintf(`{"metadata":{"annotations":{"%s":"true"}}}`, v1alpha1.AnnotationBuildRefSecret))
+
+			_, err = tb.PatchSecret(buildObject.Spec.Output.SecretRef.Name, data)
+			Expect(err).To(BeNil())
```

Comment:
Should we have an Expect that the build is still registered=False when the secret is created and only becomes True once the secret is updated with the annotation?

Reply:
@SaschaSchwarze0 Good idea!